iterative/resource_task.go

```diff
@@ -224,12 +225,22 @@ func resourceTaskDelete(ctx context.Context, d *schema.ResourceData, m interface
 func resourceTaskBuild(ctx context.Context, d *schema.ResourceData, m interface{}) (task.Task, error) {
 	v := make(map[string]*string)
 	for name, value := range d.Get("environment").(map[string]interface{}) {
-		v[name] = nil
```

Comment:
This line **is** necessary: `nil` means “import from environment” and unset just means “ignore”

Reply:
😮 Im amazed that for go its not equivalent

Reply:
### Try it with [Go Playground](https://go.dev/play/p/BAU8u30fidA)

```go
package main

import "fmt"

func main() {
	t := "text"
	m := map[string]*string{}

	m["a"] = &t
	m["b"] = nil
	// m["c"] = nil

	for k, v := range m {
		if v == nil {
			fmt.Printf("k: %s, v: %s\n", k, "<nil>")
		} else {
			fmt.Printf("k: %s, v: %s\n", k, *v)
		}
	}
}
```

### Other languages

#### JavaScript

```javascript
let object = {};

object.a = "text";
object.b = null;
// object.c = null;

for (const [key, value] of Object.entries(object)) {
  console.log(`key: ${key}, value: ${value}`);
}
```

#### Python

```python
dictionary = dict()

dictionary["a"] = "text"
dictionary["b"] = None
# dictionary["c"] = None

for key, value in dictionary.items():
    print(f"key: {key}, value: {value}")
```

Reply:
You are comparing dynamic languages. 
I thought  that go was freeing the object when pointing to nil and not that is pointing to the NULL pointer

Reply:
Even if it's an apples–to–pears comparison, all the examples are functionally equivalent. We use a `<nil>` pointer to achieve the same effect as [optional values](https://doc.rust-lang.org/std/option).

Reply:
Yes, I got it